File: WeLend Front/src/app/entities/insurance.ts
export interface Insurance{
  insuranceID:number;
  insuranceDescription:String;
  startDate:Date;
  endDate:Date;
  renewalCount:number;
  intresetRate:number;
  amount:number;

}
/*export class Insurance {
  insuranceID!:number;
  insuranceDescription!:String;
  startDate!:Date;
  endDate!:Date;
  renewalCount!:number;
  intresetRate!:number;
  amount!:number;


  constructor(insuranceID: number, insuranceDescription: String, startDate: Date, endDate: Date, renewalCount: number, intresetRate: number, amount: number) {
    this.insuranceID = insuranceID;
    this.insuranceDescription = insuranceDescription;
    this.startDate = startDate;
    this.endDate = endDate;
    this.renewalCount = renewalCount;
    this.intresetRate = intresetRate;
    this.amount = amount;
  }
}*/

File: WeLend - Copy/src/main/java/com/pidev/welend/entities/insurance.java
package com.pidev.welend.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@Table( name = "Insurance")
@JsonIgnoreProperties({"insuranceTransactions", "insuranceDetails"})
public class insurance implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="insuranceID")
    private Integer insuranceID;
    private String insuranceDescription;
    private Date startDate;
    private Date endDate;
    private Integer renewalCount;
    private double intresetRate;
    private double amount;
    @Enumerated(EnumType.STRING)
    private insuranceStatus insuranceStatus;

    @ManyToOne Account account;
    @OneToOne
    private insuranceType insuranceType;
    @OneToMany(cascade = CascadeType.ALL,mappedBy = "insurance")
    private Set<insuranceTransaction> insuranceTransactions;
    @OneToMany(cascade = CascadeType.ALL,mappedBy = "insurance")
    private Set<insuranceDetail> insuranceDetails;
}
